src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporter.cs

```diff
@@ -94,7 +96,7 @@ public override ExportResult Export(in Batch<LogRecord> logRecordBatch)
 
         try
         {
-            request.AddBatch(this.sdkLimitOptions, this.ProcessResource, logRecordBatch);
+            request.AddBatch(this.sdkLimitOptions, this.experimentalOptions, this.ProcessResource, logRecordBatch);
```

Comment:
I'm concerned about the approach here which requires adding experimentalOptions to all the functions on the call path.

Given the user will control this via environment variable `OTEL_DOTNET_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES`, I assume it is a fairly static and global thing?

I think the meta question is how we run experimental features, there seems to be multiple paths without clarity.

Reply:
`ExperimentalOptions` is added to support IConfiguration integration (not covered in this PR). 

@CodeBlanch - is there an alternate way you would be able to use to do the IConfiguration integration work you are planning?

Reply:
Right now these helper methods are all `static` extensions so everything needs to be passed around. Why not just refactor them into a class so they can access the properties/fields they need where they are needed?

```csharp
internal sealed class OtlpLogRecordProcessor
{
    private readonly SdkLimitOptions sdkLimitOptions;
    private readonly ExperimentalOptions experimentalOptions;

    public OtlpLogRecordProcessor(
       SdkLimitOptions sdkLimitOptions,
       ExperimentalOptions experimentalOptions)
    {
        this.sdkLimitOptions = sdkLimitOptions;
        this.experimentalOptions = experimentalOptions;
    }

    public OtlpCollector.ExportLogsServiceRequest BuildExportRequestForBatch(
        OtlpResource.Resource processResource,
        in Batch<LogRecord> logRecordBatch)
    {
        var request = new OtlpCollector.ExportLogsServiceRequest();

        var resourceLogs = new OtlpLogs.ResourceLogs
        {
            Resource = processResource,
        };
        request.ResourceLogs.Add(resourceLogs);

        var scopeLogs = new OtlpLogs.ScopeLogs();
        resourceLogs.ScopeLogs.Add(scopeLogs);

        foreach (var logRecord in logRecordBatch)
        {
            var otlpLogRecord = this.ConvertLogRecord(logRecord);
            if (otlpLogRecord != null)
            {
                scopeLogs.LogRecords.Add(otlpLogRecord);
            }
        }

        return request;
    }

    private OtlpLogs.LogRecord ConvertLogRecord(LogRecord logRecord)
    {
        // Access as needed:
        // this.sdkLimitOptions
        // this.experimentalOptions
    }
}
```

Change the exporter to be...

```csharp
internal sealed class OtlpLogExporter : BaseExporter<LogRecord>
{
    private readonly OtlpLogRecordProcessor logRecordProcessor;

    internal OtlpLogExporter(
        OtlpExporterOptions exporterOptions,
        SdkLimitOptions sdkLimitOptions,
        ExperimentalOptions experimentalOptions,
        IExportClient<OtlpCollector.ExportLogsServiceRequest> exportClient = null)
    {
        this.logRecordProcessor = new(sdkLimitOptions, experimentalOptions);
        // other stuff not shown
    }

    public override ExportResult Export(in Batch<LogRecord> logRecordBatch)
    {
        // Prevents the exporter's gRPC and HTTP operations from being instrumented.
        using var scope = SuppressInstrumentationScope.Begin();

        try
        {
            if (!this.exportClient.SendExportRequest(
                this.logRecordProcessor.BuildExportRequestForBatch(this.ProcessResource, in logRecordBatch))
            {
                return ExportResult.Failure;
            }
        }
        catch (Exception ex)
        {
            OpenTelemetryProtocolExporterEventSource.Log.ExportMethodException(ex);
            return ExportResult.Failure;
        }

        return ExportResult.Success;
    }
}
```

Reply:
I have done some refactoring. PTAL